projects/storefrontlib/shared/components/chat-messaging/messaging/messaging.component.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  AfterViewInit,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { WindowRef } from '@spartacus/core';
+import { ICON_TYPE } from 'projects/storefrontlib/cms-components';
+import { FilesFormValidators } from 'projects/storefrontlib/shared/services';
+import { Observable } from 'rxjs';
+import { MessageEvent, MessagingConfigs } from './messaging.model';
+
+@Component({
+  selector: 'cx-messaging',
+  templateUrl: './messaging.component.html',
+})
+export class MessagingComponent implements OnInit, AfterViewInit {
+  @Input() messageEvents$: Observable<Array<MessageEvent>>;
+  @Input() scrollToInput?: boolean = true;
+  @Input() messagingConfigs?: MessagingConfigs;
+
+  @Output()
+  send = new EventEmitter<{ files: FileList | undefined; message: string }>();
+
+  iconTypes = ICON_TYPE;
+  form: FormGroup;
+
+  MAX_INPUT_CHARACTERS: number = 2000;
+  MAX_SIZE: number = 10;
+  MAX_ENTRIES: number = 1;
+  dateFormat: string = 'MMMM d, YYYY h:mm aa';
+
+  get inputCharacterLeft(): number {
+    return (
+      (this.messagingConfigs?.charactersLimit || this.MAX_INPUT_CHARACTERS) -
+      this.form.get('message')?.value.length
+    );
+  }
+
+  get maxSize(): number {
+    return (
+      this.messagingConfigs?.attachmentRestrictions?.maxSize || this.MAX_SIZE
+    );
+  }
+
+  get maxEntries(): number {
+    return (
+      this.messagingConfigs?.attachmentRestrictions?.maxEntries ||
+      this.MAX_ENTRIES
+    );
+  }
+
+  constructor(
+    protected windowRef: WindowRef,
+    protected filesFormValidators: FilesFormValidators
+  ) {}
+
+  ngOnInit(): void {
+    this.buildForm();
+  }
+
+  ngAfterViewInit(): void {
+    if (this.scrollToInput) {
```

Comment:
let's remove this logic until it will be implemented. So let's remove ngAfterViewInit, so we can merge this as soon as  possible

Reply:
okay!